File: Programmers/코딩테스트 고득점 Kit/완전탐색/모의고사.java
import java.util.ArrayList;

/* Programmers
 * 완전탐색 - 모의고사
 * created on 2020.09.20
 * created by jionchu */

class Solution {
    public int[] solution(int[] answers) {
        int score1=0, score2=0, score3=0;
        int[][] check = {{1,2,3,4,5},
                         {2,1,2,3,2,4,2,5},
                         {3,3,1,1,2,2,4,4,5,5}};
      
        for (int i=0;i<answers.length;i++) {
            //1번 수포자
            if (answers[i] == check[0][i%5]) score1++;
            //2번 수포자
            if (answers[i] == check[1][i%8]) score2++;
            //3번 수포자
            if (answers[i] == check[2][i%10]) score3++;
      }
      
      int maxScore = Math.max(score1, Math.max(score2, score3));
      
      ArrayList<Integer> list = new ArrayList<>();
      if(maxScore == score1) {list.add(1);}
      if(maxScore == score2) {list.add(2);}
      if(maxScore == score3) {list.add(3);}
      
      return list.stream().mapToInt(i->i.intValue()).toArray();
    }
}

File: Programmers/코딩테스트 고득점 Kit/완전탐색/모의고사.kt
/* Programmers
 * 완전탐색 - 모의고사
 * created on 2021.07.04
 * created by jionchu */

class Solution {
    fun solution(answers: IntArray): IntArray {
        
        // 수포자들이 찍는 방식
        val check_first = intArrayOf(1,2,3,4,5)
        val check_second = intArrayOf(2,1,2,3,2,4,2,5)
        val check_third = intArrayOf(3,3,1,1,2,2,4,4,5,5)

        // 수포자들의 점수 배열
        val score = mutableListOf(0,0,0)

        // 정답을 맞춘 사람의 점수 증가
        for (i in 0 until answers.size) {
            if (answers[i] == check_first[i%check_first.size]) score[0]++
            if (answers[i] == check_second[i%check_second.size]) score[1]++
            if (answers[i] == check_third[i%check_third.size]) score[2]++
        }

        // 가장 높은 점수 구하기
        val max = score.max()

        // 가장 많이 맞춘 수포자들 리스트에 추가
        val answer = ArrayList<Int>()
        for (i in 0 until 3) {
            if (score[i] == max) answer.add(i+1)
        }

        return answer.toIntArray()
    }
}